src/components/charts/BarChart.tsx

```diff
@@ -0,0 +1,419 @@
+          backgroundColor: "#757575",
+        },
+      ],
+    };
+    setPrimaryData({ ...chartData });
+  }, [xAxisLabels, azurePrimaryData]);
+
+  useEffect(() => {
+    const chartData: ChartData = {
+      labels: xAxisLabels,
+      datasets: [
+        {
+          label: "Azure Secondary",
+          data: azureSecondaryData,
+          backgroundColor: "#BABABA",
+        },
+      ],
+    };
+    setSecondaryData({ ...chartData });
+  }, [xAxisLabels, azureSecondaryData]);
+
+  const getSessionData = async () => {
+    const params = {
+      period: duration,
+      starttime: startDateTime,
+      channel: channel,
+    };
+    const data = await fetchData(URL_SESSIONS, params);
+    setIsLoading(false);
+    setSessionData(data || []);
+  };
+
+  // const getHardCodedSessionData = () => {
+  //   setIsLoading(false);
+  //   setSessionData(sessionsJSON);
+  // };
+
+  const changeStartTime = (value: string) => {
+    const formattedTime = formatTime(value);
+    setStartTime(value);
+    let defaultDate: string = "";
+
+    defaultDate = !startDate
+      ? `${new Date().toISOString().split("T")[0]}`
+      : formatDate(startDate, DATE_FORMAT_2);
+
+    setStartDateTime(`${defaultDate}T${formattedTime}`);
+  };
+
+  const changeStartDate = (value: string) => {
+    setStartDate(value);
+    const selectedDate = new Date(value);
+    let defaultTime: string = "";
+    defaultTime = !startTime ? formatTime(new Date()) : formatTime(startTime);
+    setStartDateTime(
+      `${formatDate(selectedDate, DATE_FORMAT_2)}T${defaultTime}`,
+    );
+  };
+
+  const getChartConfig = (type: string) => {
+    const chartOptions = JSON.parse(JSON.stringify(BAR_CHART_OPTIONS));
+    chartOptions.plugins.title.text =
+      type === PRIMARY
+        ? `${TOTAL_SESSIONS_PER_MIN_PRIMARY}`
+        : `${TOTAL_SESSIONS_PER_MIN_SECONDARY}`;
+    chartOptions.plugins.datalabels.rotation = duration > 15 ? 270 : 0;
+    return chartOptions;
+  };
+
+  const showFilteredChannel = () => {
+    const filterApplied = CHANNEL_LIST.find((item) => item.value === channel);
+    return filterApplied ? filterApplied.label : "all";
+  };
+
+  const resetFilters = () => {
+    setStartDate("");
+    setStartTime("");
+    setDuration(5);
+    setChannel("all");
+    setSubmitCounter(0);
+  };
+
+  const incrementCounter = () => {
+    setSubmitCounter(submitCounter + 1);
+  };
+
+  const onSubmit = () => {
+    incrementCounter();
+    if (showFilterPopup) {
+      onHide(true);
+    }
+  };
+
+  const onHide = (submitClicked: boolean) => {
+    if (!submitClicked) resetFilters();
+    setShowFilterPopup(false);
+  };
+
+  const renderFooter = () => {
+    return (
+      <div>
+        <Button
+          id="popup-btn-submit"
+          className="btn-submit p-button-rounded"
+          label={SUBMIT}
+          onClick={onSubmit}
+          autoFocus
+        />
+      </div>
+    );
+  };
+
+  return (
+    <div id="bar-chart" className="m-0 p-5">
```

Comment:
I am seeing multiple nested divs.
Consider merging them.

Reply:
These divs are actually required since each of them has separate styles to handle.
For Eg. The wrapper div for input, filters, etc. are used to align the components as per the wireframe.
I don't think that it can be merged.
Please share your thoughts.